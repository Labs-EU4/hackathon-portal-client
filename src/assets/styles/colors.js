//COLORS
export const white = '#F9F9F9';
export const black = '#010038';

//RED COLOR
export const red = '#E34449';
export const brightRed = '#F34D21';

//YELLOW COLOR
export const dimYellow = '#ffdb00';

//GREY COLOR
export const grey = '#39444D';
export const textLightGrey = '#999';
export const textDarkGrey = '#666';
export const bgLightGrey = '#f0f0f0';
export const bgGrey = '#dfe3e6';
export const borderGrey = '#39444D';
// export const lightGrey = '#F5F5F5';

//GREEN COLOR
export const lightGreen = '#afeeee';
export const green = '#5aac44';
export const brightgreen = '#25d366';
export const darkGreen = '#1b9849';


// export const primary = '#04c8de'; //--> same as the above, good combo
// export const primary = '#F5F5F5'; //--> same as the above, good combo
// export const primary = '#FF3636'; //--> same as the above, good combo
// export const primary = '#47e4bb'; //--> same as the above, good combo

//PRIMARY COLOR
export const primaryLight = '#81DBFD';
export const primary = '#03AEEF';
export const primaryDark = '#03AEEF';
// export const primary = '#8CCAE9';
// export const primaryLight = '#04c8de';
// export const primary = '#40e0d0';

//SECONDARY COLOR
export const secondaryLight = '#F8C485';
export const secondary = '#f39422';
export const secondaryDark = '#DA7C0A';
// export const secondaryDark = '#F34D21';






export const backgroundBlack = '#222';
export const backgroundWhite = '#f4f4f4';
export const lightBlue = '#0ff';

// export const white = '#f4f7f6'; // --> GOOD WHITE
export const textWhite = '#F8F9FA';
export const primaryWhite = '#eff';
export const secondaryWhite = '#eee';
export const tertiaryWhite = '#ddd';
export const quaternaryWhite = '#ccc';
export const senaryWhite = '#aaa';




// export const black = '#000';
// export const black = '#1F1C05'; --> VERY GOOD BLACK SHADE
// export const black = '#020404 '; --> ALRIGHT BLACK!!!
// export const black = '#252525'; //--> SUPER GOOD BLACK
// export const black = '#202121';
export const softGreen = '#aec';
export const spaceBlue= '#03AEEF';
export const brightPurple= '#4444dd';
export const darkTwitterTheme = '#15202B';
export const lightDarkTwitterInputBg = '253341#';
export const fbCopyCatLightTheme = '#F3F4F6';
export const youTubeColorBlue = '#065FD4';
export const flickrBlue = '#1396E7';
export const btnBlue = '#0973FF';

//Add theme using styled components

//MAIN THEME COLORS
// export const primaryColor = '#BDE6EC';
// export const primaryColor = '#aec';
// export const primaryColor = '#ffdb00';
// export const primaryColor = '#6747C7'; //--> SUPER GOOD!!
// export const primaryColor = '#f8f4ff'; //--> GOOD WHITE VERSION!!
// export const primaryColor = '#fada5e'; //--> ROYAL YELLOW
// export const lightPrimaryColor = '#FEF33E'; //--> BRIGHT COOL YELLOW
// export const primaryColor = '#FFFF01'; //--> VERY BRIGHT YELLOW
// export const primaryColor = '#F4B75D'; //--> VERY BRIGHT YELLOW
// export const primaryColor = '#448ef6'; //--> VERY BRIGHT YELLOW
// export const primaryColor = '#6decb9'; //--> VERY BRIGHT YELLOW


// export const primaryColor = '#537ec5'; //--> super good color, mostly used theme!!!!
// export const turquoise = '#ff5edf'; //--> change to secondary color

export const tertiaryColor = '#5B5F97';
export const tertiaryColorDarker = '#484b77';

export const borderGrayTheme = '#39444D';

// export const primaryColor = '#1396E7'; //--> VERY GOOOD
// export const primaryColor = '#065FD4'; //--> VERY good blue, 2nd/3rd
// export const primaryColor = '#FFf'; //--> MOST USED YELLOW!!!! 
// export const primaryColor = '#B390EE';
// export const primaryColor = '#CC2D8A';
// export const primaryColor = '#46130B';
// export const primaryColor = '#FAD52D';

// export const turquoise = '#40e0d0 ';
// export const turquoise = '#4dd599'; --> VERY GOOD
// export const turquoise = '#47e4bb'; --> EVEN BETTER
// export const turquoise = '#3fc5f0';
//DARK-MODE TOGGLE COLORS
export const sunrise = '#ffc922';
export const yellowSun = '#ffdb00';
export const blueMoon = '#003366';
export const papayawhip = 'papayawhip';


// COOL COLORS 
export const newSecondaryColor = '#B390EE';
export const newTertiaryColor = '#B0EEEE';


// DARK MODE
export const MainBackgroundColor = '#29272F';
