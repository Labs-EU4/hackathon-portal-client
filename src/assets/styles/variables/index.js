export { Solid, Gradient } from './colors';
export {
    type,
    baseFontSize,
    smallFontSize,
    h1FontSize,
    h2FontSize,
    h3FontSize,
    h4FontSize,
    button_text
} from './fonts';
export { Spacing } from './spacing';
export { media } from './media';