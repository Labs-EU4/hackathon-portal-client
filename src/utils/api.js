import Axios from "axios";
import { toast } from "react-toastify";
import { resetUser } from "../store/user/actions";

const baseURL = process.env.REACT_APP_API_URL;

// if process.env.CONTEXT = "production" {
//   Point to heroku API
// }
// if process.env.BRANCH = "develop" {
  // point to Heroku staging API
// }
// if process.env.CONTEXT = "deploy-preview" {
//   point to deploy-preview HEROKU API (the separate one)
// }



export const axiosWithAuth = token =>
  Axios.create({
    baseURL,
    headers: {
      Authorization: token
    }
  });

export const axios = Axios.create({ baseURL });

export const selectToken = state => state.currentUser.token;

export const showError = message => {
  toast.error(message);
};

export const showSuccess = message => {
  toast.success(message);
};

export function* handleError({ response }, put, history) {
  const { message, statusCode } = response.data;
  if (statusCode === 404) {
    history.push("/not-found");
  }
  if (message === "jwt expired") {
    yield put(resetUser());
  } else showError(`⚠️ ${message}`);
}
