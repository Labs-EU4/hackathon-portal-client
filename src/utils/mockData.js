export const initialState = {
  events: {
    data: [
      {
        id: 1,
        event_title: "EuroHack",
        event_description: "the best event hack hack hack macbook",
        creator_id: 1,
        start_date: "10/09/11",
        end_date: "08/07/13",
        location: "Europe",
        guidelines: "cool hekhkdsjfhsdkjf skdjfhskdjfhkjsdfh skjdfhksjdfh",
        participation_type: "Team",
        category_id: 1,
        requirements: [
          {
            video_url:
              "https://res.cloudinary.com/ibkrice-com/video/upload/v1581259780/Harvesting_ifato8.mp4"
          }
        ],
        ProfileImage: [
          {
            image_url:
              "https://res.cloudinary.com/ibkrice-com/video/upload/v1581259780/Harvesting_ifato8.mp4"
          }
        ],
        rubrics: [
          "Presentation",
          "Product Fit",
          "Functionality",
          "Innovation",
          "Product Design",
          "Extensibility"
        ]
      },
      {
        id: 2,
        event_title: "World",
        event_description: "number one hacker event in the world",
        creator_id: 1,
        start_date: "07/09/11",
        end_date: "08/07/13",
        location: "Asia",
        guidelines: "come well prepared,no cheating,no fighting",
        participation_type: "Team",
        category_id: 2
      }
    ],
    categories: [
      { id: 2, category_name: "Summer Hackton" },
      { id: 3, category_name: "World Hackton" },
      { id: 3, category_name: "Asia Hackton" },
      { id: 3, category_name: "Gamer's Hackton" },
      { id: 3, category_name: "Advanced Hackton" }
    ]
  },

  currentUser: {
    token: 347534574536,
    userId: 1,
    fullname: "Jake Tom",
    email: "8omemail@google.com",
    username: "Jake22",
    bio: "I like coding ",
    image: ""
  },
  submissions: [
    {
      id: 1,
      project_title: "Mock Project",
      participant_or_team_name: "Mock Team",
      event_id: 7,
      video_url: "",
      git_url: "https://www.google.com/",
      project_writeups: "No mock write up",
      submitted_by: 14,
      created_at: "2020-03-02",
      updated_at: "2020-03-02",
      average_rating: 0,
      acted_judges: 0,
      number_of_judges: 0
    }
  ],

  teams: [
    { id: 1, team_name: "The A team", team_lead: 2 },
    { id: 2, team_name: "Hacking central", team_lead: 3 },
    { id: 3, team_name: "blackhats", team_lead: 4 },
    { id: 4, team_name: "Spies", team_lead: 1 }
  ],
  participant_team_members: [
    {
      team_member: 3,
      team_id: 1
    },
    {
      team_member: 4,
      team_id: 1
    },
    {
      team_member: 5,
      team_id: 1
    }
  ]
};
