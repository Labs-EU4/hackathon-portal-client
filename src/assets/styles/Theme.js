import * as color from './variables/colors';

const size = {
    mobileS: '320px',
    mobileM: '375px',
    mobileL: '425px',
    tablet: '768px',
    laptop: '1024px',
    laptopL: '1440px',
    desktop: '2560px'
};

export const theme = {
    color: {
        primary: {
            // light: color.primaryLight,
            regular: color.primary,
            // dark: color.primaryDark
        },
        secondary: {
            // light: color.secondaryLight,
            regular: color.secondary,
            // dark: color.secondaryDark
        },
        white: {
            regular: color.white,
        },
        black: {
            regular: color.black,
        },
        red: {
            regular: color.red,
            bright: color.brightRed
        },
        // grey: color.grey,
        grey: {
            regular: color.grey
        },
        green: {
            light: color.lightGreen,
            regular: color.green,
            dark: color.darkGreen
        },
    },
    shadow: {
        text: `text-shadow: .15px .15px 1px ${color.black}, .75px .5px 1.5px ${color.black}, .15px .15px 1px ${color.black};`,
        text_links: `1px .75px 1.5px ${color.black}`,
        box: `box-shadow: 0 3px 20px 0 rgba(0, 0, 0, .2)`,
        card: `box-shadow: 1px .5px 3px rgba(0, 0, 0, .15)`,
        intenseBox: `box-shadow: 0 2.5px 10px ${color.black};`,
        filter: `filter: drop-shadow(.5px .25px 1.5px ${color.black});
        filter: drop-shadow(.15px .15px 1px ${color.black}, .5px .25px 1.5px ${color.black}, .65px .4px 1px ${color.black});`,
        icon: `filter: drop-shadow(.5px .25px 1.5px ${color.black});
        filter: drop-shadow(25px .25px 10px ${color.black}, .5px .25px 1.5px ${color.black}, .65px .4px 1px ${color.black});`,
        btn: `2px 2px 3px ${color.black}`
    },
    flex: {
        center: 'display: flex; justify-content: center; align-items: center;',
        columnCenter: 'display: flex; flex-direction: column; justify-content: center; align-items: center;',
        column: 'display: flex; flex-direction: column;',

        custom: (justify='start', align='start', direction='row') => {
            return `
                display: flex; 
                flex-direction: ${direction}
                justify-content: ${justify}; 
                align-items: ${align};
            `
        },
    },
    device: {
        mobileS: `(min-width: ${size.mobileS})`,
        mobileM: `(min-width: ${size.mobileM})`,
        mobileL: `(min-width: ${size.mobileL})`,
        tablet: `(min-width: ${size.tablet})`,
        laptop: `(min-width: ${size.laptop})`,
        laptopL: `(min-width: ${size.laptopL})`,
        desktop: `(min-width: ${size.desktop})`,
        desktopL: `(min-width: ${size.desktop})`
    }
};