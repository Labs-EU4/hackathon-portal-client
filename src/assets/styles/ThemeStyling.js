import * as color from './variables/colors';

const size = {
    mobileS: '320px',
    mobileM: '375px',
    mobileL: '425px',
    tablet: '768px',
    laptop: '1024px',
    laptopL: '1440px',
    desktop: '2560px'
};

export const theme = {
    color: {
        primary: {
            light: '#81DBFD',
            regular: color.primary,
            // dark: color.primaryDark
        },
        secondary: {
            // light: color.secondaryLight,
            regular: color.secondary,
            // dark: color.secondaryDark
        },
        white: {
            regular: color.white,
            bg: color.bgWhite
        },
        black: {
            light: '#212121',
            regular: color.black,
        },
        red: {
            regular: color.red,
            bright: color.brightRed
        },
        blue: {
            light: color.lightBlue,
            regular: color.blue,
            bg: '#2196f3'
        },
        grey: {
            light: '#696969',
            regular: color.grey,
            dark: '#444',
            bg: color.bgGrey,
            border: 'rgba(14,19,24,.15)'
        },
        green: {
            light: color.lightGreen,
            regular: color.green,
            dark: color.darkGreen
        },
        success: color.success,
        warning: color.warning,
        danger: color.danger,
        info: color.info,
        link: {
            hover: 'rgba(14,19,24, .07)',
        }
    },
    fontSize: {
        smaller: "font-size: 1.2rem",
        small: "font-size: 1.4rem;",
        base: "font-size: 1.6rem;",
        medium: "font-size: 1.8rem",
        large: "font-size: 2rem",
        h1: "font-size: 7.5rem;",
        h2: "font-size: 5.5rem;",
        h3: "font-size: 3.5rem;",
        h4: "font-size: 2.5rem;",
        button: "font-size: 1.8rem;"
    },
    shadow: {
        text: `text-shadow: .5px .5px 1px ${color.black}`,
        text_links: `1px .75px 1.5px ${color.black}`,
        box: `box-shadow: 0 3px 20px 0 rgba(0, 0, 0, .2)`,
        card: `box-shadow: 1px .5px 3px rgba(0, 0, 0, .15)`,
        intenseBox: `box-shadow: 0 2.5px 10px ${color.black};`,
        filter: `filter: drop-shadow(.15px .15px 1.5px ${color.black});
        filter: drop-shadow(-.15px -.15px 1px ${color.black});`,
        btn: `2px 2px 3px ${color.black}`
    },
    flex: {
        center: 'display: flex; justify-content: center; align-items: center;',
        columnCenter: 'display: flex; flex-direction: column; justify-content: center; align-items: center;',
        column: 'display: flex; flex-direction: column;',

        custom: (justify='start', align='start', direction='row', wrap='no-wrap') => {
            return `
                display: flex; 
                flex-direction: ${direction}
                justify-content: ${justify}; 
                align-items: ${align};
                flex-wrap: ${wrap};
            `
        },
    },
    device: {
        mobileS: `(min-width: ${size.mobileS})`,
        mobileM: `(min-width: ${size.mobileM})`,
        mobileL: `(min-width: ${size.mobileL})`,
        tablet: `(min-width: ${size.tablet})`,
        laptop: `(min-width: ${size.laptop})`,
        laptopL: `(min-width: ${size.laptopL})`,
        desktop: `(min-width: ${size.desktop})`,
        desktopL: `(min-width: ${size.desktop})`
    }
};