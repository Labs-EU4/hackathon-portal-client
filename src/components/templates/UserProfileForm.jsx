import React, {useState} from "react";
import styled from "styled-components";
import { useHistory, Link } from "react-router-dom";
import { useDispatch } from "react-redux";
import { Formik, Form, ErrorMessage } from "formik";
import * as Yup from "yup";

import WideBody from "../atoms/WideBody";
import { H3 } from "../atoms/Heading";
import { RowHead } from "../atoms/RowHead";
import { RowBody } from "../atoms/RowBody";
import { CardWide } from "../atoms/Card";
import Label from "../atoms/Label";
import Input from "../atoms/Input";
import TextArea from "../atoms/TextArea";
import Button from "../atoms/Button";
import { media } from '../../assets/styles/variables/media';

import {
  updateUserProfile
} from "../../store/user/actions";
import { IconLetter } from "../atoms/IconLetter";

const NewLabel = styled(Label)`
  padding-left: 3px;
  @media ${media.tablet} {
    display: none;
  }
  @media ${media.mobile} {
    display: none;
  }
`;
const CardWider = styled(CardWide)`
  margin-left: 150px;
  @media ${media.tablet} {
    margin-left: 0px;
  }
  @media ${media.mobile} {
    margin-left: 0px;
  }
`;
const ButtonRowBody = styled(RowBody)`
@media ${media.tablet} {
  justify-content: space-around;
}
`
const NewButton = styled(Button)`
@media ${media.tablet} {
  width: 25%;
}
@media ${media.mobile} {
  width: 50%
}
`

const UserProfileForm = ({ initialState, isProfileOpen, setIsProfileOpen }) => {
  const dispatch = useDispatch();
  const history = useHistory();
  const [selectedImage, setSelectedImage] = useState(initialState?.image_url);

  const handleSubmit = (values, a) => {
    const formData = new FormData();
    formData.append('image_url', selectedImage);
    formData.append('bio', values.bio);
    formData.append('fullname', values.fullname);
    formData.append('email', values.email);
    formData.append('username', values.username);
    dispatch(updateUserProfile(formData, history));
    setIsProfileOpen(false);
  };

  const defaultState = {
    bio: initialState?.bio || "",
    fullname: initialState?.fullname || "",
    email: initialState?.email || "",
    username: initialState?.username || ""
  }

  const schema = Yup.object().shape({
    fullname: Yup.string().required("fullname is required"),
    email: Yup.string().required("email is required"),
    username: Yup.string().required("username is required"),
    bio: Yup.string()
  });

  return (
    <StyledWideBody active={isProfileOpen}>
      <Formik
        onSubmit={handleSubmit}
        initialValues={defaultState}
        validationSchema={schema}
        enableReinitialize
      >
        {({ errors, touched }) => (
          <Form>
            <RowHead center bold>
              <H3>Edit Profile</H3>
            </RowHead>
            <RowBody>
              <Label htmlFor="fullname">Full Name</Label>
              <Input
                type="text"
                name="fullname"
                display="wide"
                placeholder="Full Name"
              />
            </RowBody>
            <RowBody>
              <Label htmlFor="username">Username</Label>
              <Input
                type="text"
                name="username"
                placeholder="Username"
                display="wide"
              />
              {errors.name && touched.name ? (
                <div>{errors.name}</div>
              ) : null}
              <ErrorMessage name="username" />
            </RowBody>
            <RowBody>
              <Label htmlFor="email">Email</Label>
              <Input
                type="text"
                name="email"
                placeholder="Email"
                display="wide"
              />
              {errors.name && touched.name ? (
                <div>{errors.name}</div>
              ) : null}
              <ErrorMessage name="email" />
            </RowBody>
            <RowBody>
              <Label htmlFor="image">Profile Image</Label>
              <Input
                type="file"
                name="image_url"
                display="wide"
                placeholder="Profile picture"
                accept="image/*"
                onChange={(e) => setSelectedImage(e.target.files[0])}
              />
            </RowBody>
            <RowBody>
              <Label htmlFor="bio">Bio</Label>
              <TextArea
                wide
                tall
                as="textarea"
                name="bio"
                placeholder="bio"
              />
              {errors.name && touched.name ? (
                <div>{errors.name}</div>
              ) : null}
              <ErrorMessage name="bio" />
            </RowBody>
            <ButtonRowBody>
              {/* <Link to="/dashboard">
                <Button to="/dashboard" color="grey">
                  Cancel
                </Button>
              </Link> */}
              <NewButton 
                color="green" 
                size="wide"
                type="submit"
              >Save Changes</NewButton>
            </ButtonRowBody>
          </Form>
        )}
      </Formik>
    </StyledWideBody>
  );
};

export default UserProfileForm;


const StyledWideBody = styled(WideBody)`
  ${props => props.theme.shadow.box};
  position: absolute; top: 0; right: 0;
  width: 300px; height: 100%;
  background-color: ${props => props.theme.color.white.bg};
  border: 2px solid  ${props => props.theme.color.primary.regular};
  border-right: none;
  transform: ${props => !props.active &&'translateX(100%)'};
  transition: transform 1s ease;
  padding: 20px;
`;
