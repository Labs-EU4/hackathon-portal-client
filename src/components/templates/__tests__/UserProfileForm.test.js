import React from "react";
import { Router } from "react-router";
import { createMemoryHistory } from "history";
import { Provider } from "react-redux";
import { render, cleanup } from "@testing-library/react";
import configureStore from "redux-mock-store";
import "@testing-library/jest-dom/extend-expect";
import UserProfileForm from "../UserProfileForm";
import { initialState } from "../../../utils/mockData";
import { theme } from "../../../assets/styles/ThemeStyling";
import { ThemeProvider } from "styled-components";
import { library } from '@fortawesome/fontawesome-svg-core';
import { fas } from '@fortawesome/free-solid-svg-icons';
import { far } from '@fortawesome/free-regular-svg-icons';
import { fab } from '@fortawesome/free-brands-svg-icons';

library.add(fas, far, fab);

const history = createMemoryHistory();

afterEach(cleanup);

let jestFeatures;
let mockStore;
let store;

beforeEach(() => {
  mockStore = configureStore();
  store = mockStore(initialState);
  jestFeatures = render(
    <Router history={history}>
      <Provider store={store}>
        <ThemeProvider theme={theme}>
          <UserProfileForm initialState={initialState} />
        </ThemeProvider>
      </Provider>
    </Router>
  );
});


describe("Renders Without Crashing", () => {
  it("Renders Without Crashing.", () => {
    expect(jestFeatures).toMatchSnapshot();
  });
  it("should be displaying the label text node for the Heading Edit Profile", () => {
    let mainHeader = () => jestFeatures.getByText("Edit Profile");
    expect(mainHeader()).toBeInTheDocument();
  });
});
