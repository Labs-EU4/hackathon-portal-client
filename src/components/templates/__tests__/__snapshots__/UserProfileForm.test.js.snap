// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders Without Crashing Renders Without Crashing. 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <div
          class="sc-bdVaJa gOhKpA"
        >
          <div
            class="sc-htpNat dAVxPZ"
          >
            <a
              href="/dashboard"
            >
              <figure
                class="sc-EHOje evAtGa"
              >
                <p>
                  HackHunt
                </p>
              </figure>
            </a>
            <div
              class="sc-kafWEX cVkhK"
            >
              <div
                class="sc-EHOje kXozTT"
              >
                8
                <ul
                  class="sc-bZQynM fVGedF"
                >
                  <li>
                    <a
                      href="/dashboard/profile/edit"
                    >
                      Settings
                    </a>
                  </li>
                  <li>
                    <span>
                      LogOut
                    </span>
                  </li>
                </ul>
              </div>
            </div>
          </div>
        </div>
        <div
          class="sc-jhAzac jsMUOz"
        >
          <div
            class="sc-fMiknA heulbj sc-bbmXgH cMYzZi"
          >
            <div
              class="sc-gisBJw fdfqLi"
            >
              <h3
                class="sc-eNQAEJ faMTgL"
              >
                Edit Profile
              </h3>
            </div>
            <div
              class="sc-cHGsZl eIVmdX"
            >
              <div
                class="sc-kvZOFW sc-hqyNC sc-jnlKLf chWpoq"
              >
                <form
                  action="#"
                >
                  <label
                    class="sc-fAjcbJ sc-gGBfsJ fVseha"
                    for="image"
                  >
                    Profile picture
                  </label>
                  <figure
                    class="sc-frDJqD ycfaW"
                  >
                    <div>
                      <img
                        alt=""
                        src="profile-image.png"
                      />
                    </div>
                    <p />
                  </figure>
                  <div
                    class="sc-cHGsZl eIVmdX"
                  >
                    <label
                      class="sc-fAjcbJ bGOVdV"
                      for="fullname"
                    >
                      Full Name
                    </label>
                    <input
                      class="sc-eHgmQL cwnFTa"
                      display="wide"
                      name="fullname"
                      placeholder="Full Name"
                      type="text"
                      value=""
                    />
                  </div>
                  <div
                    class="sc-cHGsZl eIVmdX"
                  >
                    <label
                      class="sc-fAjcbJ bGOVdV"
                      for="image"
                    >
                      Profile Image
                    </label>
                    <input
                      accept="image/*"
                      class="sc-eHgmQL cwnFTa"
                      display="wide"
                      name="image_url"
                      placeholder="Profile picture"
                      type="file"
                      value=""
                    />
                  </div>
                  <div
                    class="sc-cHGsZl eIVmdX"
                  >
                    <div
                      class="sc-kgAjT fhOGzY"
                    >
                      <label
                        class="sc-fAjcbJ bGOVdV"
                        for="email"
                      >
                        Email
                      </label>
                      <input
                        class="sc-eHgmQL cwnFTa"
                        display="wide"
                        name="email"
                        placeholder="Email"
                        type="text"
                        value=""
                      />
                    </div>
                    <div
                      class="sc-kgAjT fhOGzY"
                    >
                      <label
                        class="sc-fAjcbJ bGOVdV"
                        for="username"
                      >
                        Username
                      </label>
                      <input
                        class="sc-eHgmQL cwnFTa"
                        display="wide"
                        name="username"
                        placeholder="Username"
                        type="text"
                        value=""
                      />
                    </div>
                  </div>
                  <div
                    class="sc-cHGsZl eIVmdX"
                  >
                    <label
                      class="sc-fAjcbJ bGOVdV"
                      for="bio"
                    >
                      Bio
                    </label>
                    <div
                      class="sc-ksYbfQ colGcy"
                    >
                      <textarea
                        name="bio"
                        placeholder="bio"
                      />
                    </div>
                  </div>
                  <div
                    class="sc-cHGsZl sc-fYxtnH jmsDoI"
                  >
                    <a
                      href="/dashboard"
                    >
                      <button
                        class="sc-cvbbAY ebepNj"
                        color="grey"
                        to="/dashboard"
                      >
                        Cancel
                      </button>
                    </a>
                    <button
                      class="sc-cvbbAY hdTOaT sc-tilXH jSiPSU"
                      color="green"
                      type="submit"
                    >
                      Save Changes
                    </button>
                  </div>
                </form>
              </div>
            </div>
          </div>
        </div>
        <div
          class="sc-gzVnrw bRGEaa"
        >
          <div
            class="sc-dnqmqq dBtDWt"
          >
            <div>
              <img
                alt="LinkedIn"
                class="sc-gZMcBi fKNoMs"
                src="Icon-linkedin.png"
              />
              <img
                alt="Twitter"
                class="sc-gZMcBi fKNoMs"
                src="Icon-twitter.png"
              />
              <img
                alt="Facebook"
                class="sc-gZMcBi fKNoMs"
                src="Icon-facebook.png"
              />
            </div>
            <div>
              <a
                class="sc-gqjmRU sc-jTzLTM fddzAd"
                href="https://hackton.co/#team"
              >
                Team
              </a>
              <a
                class="sc-gqjmRU sc-jTzLTM fddzAd"
                href="#"
              >
                Privacy
              </a>
              <a
                class="sc-gqjmRU sc-jTzLTM fddzAd"
                href="#"
              >
                Contacts
              </a>
            </div>
            <div>
              <p
                class="sc-kAzzGY sc-chPdSV gAkGJE"
              >
                International Crafters Â© 2020
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <div
        class="sc-bdVaJa gOhKpA"
      >
        <div
          class="sc-htpNat dAVxPZ"
        >
          <a
            href="/dashboard"
          >
            <figure
              class="sc-EHOje evAtGa"
            >
              <p>
                HackHunt
              </p>
            </figure>
          </a>
          <div
            class="sc-kafWEX cVkhK"
          >
            <div
              class="sc-EHOje kXozTT"
            >
              8
              <ul
                class="sc-bZQynM fVGedF"
              >
                <li>
                  <a
                    href="/dashboard/profile/edit"
                  >
                    Settings
                  </a>
                </li>
                <li>
                  <span>
                    LogOut
                  </span>
                </li>
              </ul>
            </div>
          </div>
        </div>
      </div>
      <div
        class="sc-jhAzac jsMUOz"
      >
        <div
          class="sc-fMiknA heulbj sc-bbmXgH cMYzZi"
        >
          <div
            class="sc-gisBJw fdfqLi"
          >
            <h3
              class="sc-eNQAEJ faMTgL"
            >
              Edit Profile
            </h3>
          </div>
          <div
            class="sc-cHGsZl eIVmdX"
          >
            <div
              class="sc-kvZOFW sc-hqyNC sc-jnlKLf chWpoq"
            >
              <form
                action="#"
              >
                <label
                  class="sc-fAjcbJ sc-gGBfsJ fVseha"
                  for="image"
                >
                  Profile picture
                </label>
                <figure
                  class="sc-frDJqD ycfaW"
                >
                  <div>
                    <img
                      alt=""
                      src="profile-image.png"
                    />
                  </div>
                  <p />
                </figure>
                <div
                  class="sc-cHGsZl eIVmdX"
                >
                  <label
                    class="sc-fAjcbJ bGOVdV"
                    for="fullname"
                  >
                    Full Name
                  </label>
                  <input
                    class="sc-eHgmQL cwnFTa"
                    display="wide"
                    name="fullname"
                    placeholder="Full Name"
                    type="text"
                    value=""
                  />
                </div>
                <div
                  class="sc-cHGsZl eIVmdX"
                >
                  <label
                    class="sc-fAjcbJ bGOVdV"
                    for="image"
                  >
                    Profile Image
                  </label>
                  <input
                    accept="image/*"
                    class="sc-eHgmQL cwnFTa"
                    display="wide"
                    name="image_url"
                    placeholder="Profile picture"
                    type="file"
                    value=""
                  />
                </div>
                <div
                  class="sc-cHGsZl eIVmdX"
                >
                  <div
                    class="sc-kgAjT fhOGzY"
                  >
                    <label
                      class="sc-fAjcbJ bGOVdV"
                      for="email"
                    >
                      Email
                    </label>
                    <input
                      class="sc-eHgmQL cwnFTa"
                      display="wide"
                      name="email"
                      placeholder="Email"
                      type="text"
                      value=""
                    />
                  </div>
                  <div
                    class="sc-kgAjT fhOGzY"
                  >
                    <label
                      class="sc-fAjcbJ bGOVdV"
                      for="username"
                    >
                      Username
                    </label>
                    <input
                      class="sc-eHgmQL cwnFTa"
                      display="wide"
                      name="username"
                      placeholder="Username"
                      type="text"
                      value=""
                    />
                  </div>
                </div>
                <div
                  class="sc-cHGsZl eIVmdX"
                >
                  <label
                    class="sc-fAjcbJ bGOVdV"
                    for="bio"
                  >
                    Bio
                  </label>
                  <div
                    class="sc-ksYbfQ colGcy"
                  >
                    <textarea
                      name="bio"
                      placeholder="bio"
                    />
                  </div>
                </div>
                <div
                  class="sc-cHGsZl sc-fYxtnH jmsDoI"
                >
                  <a
                    href="/dashboard"
                  >
                    <button
                      class="sc-cvbbAY ebepNj"
                      color="grey"
                      to="/dashboard"
                    >
                      Cancel
                    </button>
                  </a>
                  <button
                    class="sc-cvbbAY hdTOaT sc-tilXH jSiPSU"
                    color="green"
                    type="submit"
                  >
                    Save Changes
                  </button>
                </div>
              </form>
            </div>
          </div>
        </div>
      </div>
      <div
        class="sc-gzVnrw bRGEaa"
      >
        <div
          class="sc-dnqmqq dBtDWt"
        >
          <div>
            <img
              alt="LinkedIn"
              class="sc-gZMcBi fKNoMs"
              src="Icon-linkedin.png"
            />
            <img
              alt="Twitter"
              class="sc-gZMcBi fKNoMs"
              src="Icon-twitter.png"
            />
            <img
              alt="Facebook"
              class="sc-gZMcBi fKNoMs"
              src="Icon-facebook.png"
            />
          </div>
          <div>
            <a
              class="sc-gqjmRU sc-jTzLTM fddzAd"
              href="https://hackton.co/#team"
            >
              Team
            </a>
            <a
              class="sc-gqjmRU sc-jTzLTM fddzAd"
              href="#"
            >
              Privacy
            </a>
            <a
              class="sc-gqjmRU sc-jTzLTM fddzAd"
              href="#"
            >
              Contacts
            </a>
          </div>
          <div>
            <p
              class="sc-kAzzGY sc-chPdSV gAkGJE"
            >
              International Crafters Â© 2020
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
